src/docs/guides/23-how-to-use-custom-pricing/index.md

```diff
@@ -0,0 +1,793 @@
+checkbox form value into the actual cleaning fee of this listing:
+
+```js
+const handleSubmit = values => {
+  const selectedCleaningFee =
+    values &&
+    values.additionalItems &&
+    values.additionalItems[0] === 'cleaningFee'
+      ? cleaningFee
+      : null;
+  onSubmit({
+    ...values,
+    cleaningFee: selectedCleaningFee,
+  });
+};
+```
+
+Pass `cleaningFee` and `handleSubmit` to the `BookingDatesForm`
+component as follows:
+
+```js
+<BookingDatesForm
+  className={css.bookingForm}
+  submitButtonWrapperClassName={css.bookingDatesSubmitButtonWrapper}
+  unitType={unitType}
+  onSubmit={handleSubmit}
+  price={price}
+  isOwnListing={isOwnListing}
+  timeSlots={timeSlots}
+  fetchTimeSlotsError={fetchTimeSlotsError}
+  cleaningFee={cleaningFee}
+/>
+```
+
+In the `BookingDatesForm` component file import the `FieldCheckbox`
+component and the `formatMoney` utility function:
+
+```js
+import {
+  Form,
+  PrimaryButton,
+  FieldDateRangeInput,
+  FieldCheckbox,
+} from '../../components';
+import { formatMoney } from '../../util/currency';
+```
+
+Add a new entry to the component props map: `cleaningFee`. It holds the
+price of the listing's cleaning fee in a `Money` object. Also we need to
+resolve the current value for an `additionalItems` input in the form and
+a label from the translations so that we can add an input for cleaning
+fee to the form.
+
+To resolve the current cleaning fee selection from the form, add the
+following lines to the `render` funciton of the `FinalForm` element in
+`BookingDatesForm`:
+
+```js
+const selectedCleaningFee =
+  values &&
+  values.additionalItems &&
+  values.additionalItems[0] === 'cleaningFee'
+    ? cleaningFee
+    : null;
+```
+
+Add the resolved `selectedCleaningFee` as a `cleaningFee` attribute to
+the bookingData object:
+
+```js
+const bookingData =
+  startDate && endDate
+    ? {
+        unitType,
+        unitPrice,
+        startDate,
+        endDate,
+
+        // NOTE: If unitType is `line-item/units`, a new picker
+        // for the quantity should be added to the form.
+        quantity: 1,
+        cleaningFee: selectedCleaningFee,
+      }
+    : null;
+```
+
+Also add the following line inside the `BookingDatesForm` component
+before the `return` statement:
+
+```js
+const cleaningFeeLabel = intl.formatMessage({
+  id: 'BookingDatesForm.cleaningFee',
+});
+```
+
+Now we can add an input for selecting a cleaning fee to be added to a
+booking, for example right after the `FieldDateRangeInput` inside the
+`Form` element:
+
+```js
+{
+  cleaningFee ? (
+    <div className={css.cleaningFee}>
+      <FieldCheckbox
+        className={css.cleaningFeeLabel}
+        id={`${form}.cleaningFee`}
+        label={cleaningFeeLabel}
+        name={'additionalItems'}
+        value={'cleaningFee'}
+      />
+      <span className={css.cleaningFeeAmount}>
+        {formatMoney(intl, cleaningFee)}
+      </span>
+    </div>
+  ) : null;
```

Comment:
Has this gone through Prettier?
I think it strips off semicolons if they are inside returned JSX.

Reply:
That's true, but the docs formatting unfolds it like this.

Reply:
Added a prettier ignore here.